go/private/toolchain.bzl

```diff
@@ -37,20 +72,24 @@ def _go_sdk_impl(ctx):
     _sdk_build_file(ctx, str(ctx.path(".")))
     _remote_sdk(ctx, urls, ctx.attr.strip_prefix, ctx.attr.sha256)
   elif ctx.attr.path:
+    print("DEPRECATED: go_sdk with a path, please use go_local_sdk")
     _sdk_build_file(ctx, ctx.attr.path)
     _local_sdk(ctx, ctx.attr.path)
   else:
+    print("DEPRECATED: go_sdk without path or urls, please use go_host_sdk")
     path = _detect_host_sdk(ctx)
     _sdk_build_file(ctx, path)
     _local_sdk(ctx, path)
+  _prepare(ctx)
 
+
+def _prepare(ctx):
   if "TMP" in ctx.os.environ:
     tmp = ctx.os.environ["TMP"]
     ctx.symlink(tmp, "tmp")
   else:
     ctx.file("tmp/ignore", content="") # make a file to force the directory to exist
     tmp = str(ctx.path("tmp").realpath)
-
   # Build the standard library for valid cross compile platforms
   #TODO: fix standard library cross compilation
   if ctx.name.endswith("linux_amd64") and ctx.os.name == "linux":
```

Comment:
I think this PR broke this section - `ctx.name` always seems to be `go_sdk` now, so none of these conditionals match.